torch/csrc/jit/codegen/cuda/lower_loops.cpp

```diff
@@ -90,100 +90,154 @@ void LoopNestGenerator::pushFront(kir::Expr* expr) {
   }
 }
 
-void LoopNestGenerator::handle(const Expr* expr) {
-  const auto gpu_lower = GpuLower::current();
-  kir::IrBuilder ir_builder(gpu_lower->kernel());
+// Generate the loop nest structure and place it in lowered_exprs_
+void LoopNestGenerator::generate(const std::vector<Expr*>& exprs) {
+  TORCH_INTERNAL_ASSERT(lowered_exprs_.empty());
 
-  // Check if it's a tensor view expression we need to place in the loop nest
-  // structure
-  if (!ir_utils::isTVOp(expr)) {
-    // Close all the loops, scalar operations cannot be inside for loops based
-    // on expr sorting.
-    while (!for_loops_.empty()) {
-      closeFor();
-    }
-    pushFront(gpu_lower->lowerExpr(expr));
+  // How many loops can the next iteration close
+  std::ptrdiff_t min_close = -1;
```

Comment:
IIUC, wouldn't `max_close` better reflect what it actually means? It seems this is used to control the maximum number of loops that could be closed.